modules/util/AsyncQueue.js

```diff
@@ -31,6 +48,8 @@ export default class AsyncQueue {
         } catch (error) {
             logger.error(`Task failed: ${error?.stack}`);
             finishedCallback(error);
```

Comment:
You need to try-catch here too.

Reply:
I don't understand why, an error during the callback here would not interrupt anything (there's nothing to process afterwards). Isn't it better to let the error bubble up instead of just logging it?

Reply:
Ah right, sorry I hadn't looked at the whole context.